### PR TITLE
Replace String.CompareOrdinal to string.Equals Final Part

### PR DESCRIPTION
**Pull request by lindexi**
## Description

We can use string.Equals(string1, string2, StringComparison.Ordinal) to check for equality, see https://learn.microsoft.com/en-us/dotnet/csharp/how-to/compare-strings and https://learn.microsoft.com/en-us/dotnet/standard/base-types/best-practices-strings

> Use the [String.Compare](https://learn.microsoft.com/en-us/dotnet/api/system.string.compare) and [String.CompareTo](https://learn.microsoft.com/en-us/dotnet/api/system.string.compareto) methods to sort strings, not to check for equality.

Reference: https://github.com/dotnet/wpf/issues/7834

## Customer Impact

Small performance improvements

## Regression

None.

## Testing

Just CI.

## Risk

Low.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/wpf/pull/8042)

### CODE REVIEW

File: src/Microsoft.DotNet.Wpf/src/WindowsBase/System/Security/RightsManagement/User.cs [hunk 1]
```diff
@@ -135,7 +135,7 @@ public override bool Equals(object obj)
 
             ContentUser userObj = (ContentUser)obj;
 
-            return (String.CompareOrdinal(_name.ToUpperInvariant(), userObj._name.ToUpperInvariant()) == 0)
```
Comment (rchauhan18) [inline src/Microsoft.DotNet.Wpf/src/WindowsBase/System/Security/RightsManagement/User.cs:138]: @lindexi Did you intentionally left "ToUpperInvariant()" here?
If no, then we can update it.

Comment (lindexi) [reply src/Microsoft.DotNet.Wpf/src/WindowsBase/System/Security/RightsManagement/User.cs:138]: Thank you @rchauhan18 and I will fix it.

**Comment by rchauhan18** [comment]
@lindexi, thanks for your contribution

**Comment by lindexi** [comment]
Thank you @rchauhan18

**Pull request merged**